### PR TITLE
feat(route): add 山青学生在线、山大研工部、山大国际网站

### PR DESCRIPTION
**Pull request by kukeya**
## Involved Issue / 该 PR 相关 Issue

Close #

## Example for the Proposed Route(s) / 路由地址示例

```routes
/sdu/gjsw/tzgg
/sdu/qd/xszxqd/xtyw
/sdu/qd/xszxqd/xttz-yjs
/sdu/qd/xszxqd/xttz-bks
/sdu/qd/xszxqd/xttz-tx
/sdu/qd/xszxqd/xttz-xl
/sdu/ygb/zytz
/sdu/ygb/glfw
/sdu/ygb/cxsj
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [x] New Route / 新的路由
  - [x] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Anti-bot or rate limit / 反爬/频率限制
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [x] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [x] Parsed / 可以解析
  - [ ] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

## Note / 说明
`/sdu/qd/xszxqd/`需校园内网环境访问，经测试内网部署通过

### CODE REVIEW

**Review by wiketool — APPROVED** [review]

File: lib/routes/sdu/gjsw.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    name: '山东大学国际事务部',
+    maintainers: ['kukeya'],
+    handler,
+    description: `| 通知公告 |  
+  | -------- | 
+  | tzgg     |      `,
+};
+
+async function handler(ctx) {
+    const type = ctx.req.param('type') ? ctx.req.param('type') : 'tzgg';
```
Comment (wiketool) [inline lib/routes/sdu/gjsw.ts:39]:
```
ctx.req.param('type') ?? 'tzgg';
```
more better

File: lib/routes/sdu/gjsw.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    const link = new URL(typeMap[type].url, host).href;
+
+    const response = await got(link);
+
+    const $ = load(response.data);
+
+    let item = $('.dqlb ul li')
```
Comment (wiketool) [inline lib/routes/sdu/gjsw.ts:46]:
```
let item = $('.dqlb ul li').toArray()
```

File: lib/routes/sdu/gjsw.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    parameters: { type: '默认为`tzgg`' },
+    features: {
+        requireConfig: false,
+        requirePuppeteer: false,
+        antiCrawler: false,
+        supportBT: false,
+        supportPodcast: false,
+        supportScihub: false,
+    },
+    name: '山东大学国际事务部',
```
Comment (TonyRL) [inline lib/routes/sdu/gjsw.ts:30]: No need to repeat namespace's name in route name.

File: lib/routes/sdu/ygb.ts [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    parameters: { type: '默认为`zytz`' },
+    features: {
+        requireConfig: false,
+        requirePuppeteer: false,
+        antiCrawler: false,
+        supportBT: false,
+        supportPodcast: false,
+        supportScihub: false,
+    },
+    name: '山东大学研工部',
```
Comment (TonyRL) [inline lib/routes/sdu/ygb.ts:38]: No need to repeat namespace's name in route name.

File: lib/routes/sdu/ygb.ts [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+    const $ = load(response.data);
+
+    let item = $('.zytz-list li')
+        .map((_, e) => {
+            e = $(e);
+            const a = e.find('a');
+            let link = '';
+            link = a.attr('href').startsWith('info/') || a.attr('href').startsWith('content') ? host + a.attr('href') : a.attr('href');
+            return {
+                title: a.text().trim(),
```
Comment (TonyRL) [inline lib/routes/sdu/ygb.ts:61]: Some feed entries of `/sdu/ygb/zytz` contain empty title
https://github.com/DIYgod/RSSHub/pull/16812#issuecomment-2360669083

File: lib/routes/sdu/ygb.ts [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+                    return finishArticleItem(item);
+                } else if (new URL(item.link).hostname !== 'www.ygb.sdu.edu.cn') {
+                    return item;
+                }
+                const response = await got(item.link);
+                const $ = load(response.data);
+
+                item.title = $('.article-tlt').text().trim();
+                $('.article-tlt').remove();
+                item.description = $('form[name=_newscontent_fromname]').html();
```
Comment (TonyRL) [inline lib/routes/sdu/ygb.ts:81]: Do not include `author` and `pubDate` in the `description`

File: lib/routes/sdu/qd/xszxqd.ts [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+
+    const $ = load(response.data);
+
+    let item = $('.list_box li')
+        .toArray()
+        .map((e) => {
+            e = $(e);
+            const a = e.find('a');
+            let link = '';
+            link = a.attr('href').startsWith('tz_content') || a.attr('href').startsWith('content') ? host + a.attr('href') : a.attr('href');
```
Comment (TonyRL) [inline lib/routes/sdu/qd/xszxqd.ts:68]:
```suggestion
            const link = a.attr('href').startsWith('tz_content') || a.attr('href').startsWith('content') ? host + a.attr('href') : a.attr('href');
```
`link` is not reassigned afterwards

File: lib/routes/sdu/ygb.ts [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+    const response = await got(link);
+
+    const $ = load(response.data);
+    let item = $('.zytz-list li')
+        .toArray()
+        .map((e) => {
+            e = $(e);
+            const a = e.find('a');
+            let link = '';
+            link = a.attr('href').startsWith('info/') || a.attr('href').startsWith('content') ? host + a.attr('href') : a.attr('href');
```
Comment (TonyRL) [inline lib/routes/sdu/ygb.ts:59]:
```suggestion
            const link = a.attr('href').startsWith('info/') || a.attr('href').startsWith('content') ? host + a.attr('href') : a.attr('href');
```

**Pull request merged**